lms/static/scripts/frontend_apps/utils/google-picker-client.js

```diff
@@ -76,15 +80,53 @@ export class GooglePickerClient {
    * @return {Promise<string>} - An access token for making Google Drive API requests.
    */
   async _authorizeDriveAccess() {
-    const auth2 = await this._gapiAuth2;
-    const googleAuth = auth2.init({
+    const idServices = await this._identityServices;
+
+    /** @type {(err: Error) => void} */
+    let rejectAccessToken;
+
+    /** @type {(token: string) => void} */
+    let resolveAccessToken;
+    const accessToken = new Promise((resolve, reject) => {
+      resolveAccessToken = resolve;
+      rejectAccessToken = reject;
+    });
+
+    // See https://developers.google.com/identity/oauth2/web/reference/js-reference#CodeClientConfig.
+    const client = idServices.accounts.oauth2.initTokenClient({
       client_id: this._clientId,
       scope: GOOGLE_DRIVE_SCOPE,
+
+      // Callback used for successful responses and OAuth errors.
```

Comment:
Slightly confusingly, the GIS library can report errors via two different callbacks, depending on whether the error was a failed server call or a local error (eg. failed to open popup window).